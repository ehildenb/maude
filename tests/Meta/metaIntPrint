#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude \
  < $srcdir/metaIntPrint.maude -no-banner -no-advise \
  > metaIntPrint.out 2>&1

diff $srcdir/metaIntPrint.expected metaIntPrint.out > /dev/null 2>&1
