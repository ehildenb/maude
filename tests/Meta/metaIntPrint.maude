set show timing off .
set show advisories off .

***
*** Test pretty printing in the meta-interpreter.
***

load metaInterpreter

mod PRINT-TEST is
  pr META-INTERPRETER .
  pr RAT .

  op me : -> Oid .
  op User : -> Cid .
  op term:_ : Term -> Attribute .
  op flags:_ : PrintOptionSet  -> Attribute .

  vars X Y Z : Oid .
  var AS : AttributeSet .
  var T : Term .
  var F : PrintOptionSet .

  rl < X : User | AS > createdInterpreter(X, Y, Z) =>
     < X : User | AS > insertModule(Z, X, upModule('RAT, true)) .

  rl < X : User | AS, term: T, flags: F > insertedModule(X, Y) =>
     < X : User | AS > printTerm(Y, X, 'RAT, T, F) .
endm

erew in PRINT-TEST : <> 
< me : User | term: upTerm(1/2), flags: mixfix > 
createInterpreter(interpreterManager, me, none) .

erew in PRINT-TEST : <> 
< me : User | term: upTerm(1/2), flags:  number > 
createInterpreter(interpreterManager, me, none) .

erew in PRINT-TEST : <> 
< me : User | term: upTerm(1/2), flags: rat > 
createInterpreter(interpreterManager, me, none) .

erew in PRINT-TEST : <> 
< me : User | term: upTerm(1/2 + X:Rat), flags: rat > 
createInterpreter(interpreterManager, me, none) .

erew in PRINT-TEST : <> 
< me : User | term: upTerm(1/2 + X:Rat), flags: (rat mixfix) > 
createInterpreter(interpreterManager, me, none) .

erew in PRINT-TEST : <> 
< me : User | term: upTerm(1/2 + X:Rat), flags: (number mixfix) > 
createInterpreter(interpreterManager, me, none) .

erew in PRINT-TEST : <> 
< me : User | term: upTerm(1/2 + X:Rat), flags: mixfix > 
createInterpreter(interpreterManager, me, none) .
