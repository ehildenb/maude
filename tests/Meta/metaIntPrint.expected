==========================================
erewrite in PRINT-TEST : <> < me : User | term: upTerm(1/2),flags: mixfix >
    createInterpreter(interpreterManager, me, none) .
rewrites: 6
result Configuration: <> < me : User | none > printedTerm(me, interpreter(0),
    's '0 '/ 's_^2 '`( '0 '`))
==========================================
erewrite in PRINT-TEST : <> < me : User | term: upTerm(1/2),flags: number >
    createInterpreter(interpreterManager, me, none) .
rewrites: 6
result Configuration: <> < me : User | none > printedTerm(me, interpreter(0),
    '_/_ '`( '1 '`, '2 '`))
==========================================
erewrite in PRINT-TEST : <> < me : User | term: upTerm(1/2),flags: rat >
    createInterpreter(interpreterManager, me, none) .
rewrites: 6
result Configuration: <> < me : User | none > printedTerm(me, interpreter(0),
    '1/2)
==========================================
erewrite in PRINT-TEST : <> < me : User | term: upTerm(X:Rat + 1/2),flags: rat
    > createInterpreter(interpreterManager, me, none) .
rewrites: 6
result Configuration: <> < me : User | none > printedTerm(me, interpreter(0),
    '_+_ '`( 'X:Rat '`, '1/2 '`))
==========================================
erewrite in PRINT-TEST : <> < me : User | term: upTerm(X:Rat + 1/2),flags: (
    mixfix rat) > createInterpreter(interpreterManager, me, none) .
rewrites: 6
result Configuration: <> < me : User | none > printedTerm(me, interpreter(0),
    'X:Rat '+ '1/2)
==========================================
erewrite in PRINT-TEST : <> < me : User | term: upTerm(X:Rat + 1/2),flags: (
    mixfix number) > createInterpreter(interpreterManager, me, none) .
rewrites: 6
result Configuration: <> < me : User | none > printedTerm(me, interpreter(0),
    'X:Rat '+ '1 '/ '2)
==========================================
erewrite in PRINT-TEST : <> < me : User | term: upTerm(X:Rat + 1/2),flags:
    mixfix > createInterpreter(interpreterManager, me, none) .
rewrites: 6
result Configuration: <> < me : User | none > printedTerm(me, interpreter(0),
    'X:Rat '+ 's '0 '/ 's_^2 '`( '0 '`))
Bye.
